### ISSUE TITLE
Fails on setup because port is in use

### ISSUE THREAD

**Issue by sophof** [home-assistant-libs/aioslimproto#39]
Tried to install this integration to try out the new music assistant. Installed squeezelite on a raspberry pi. When adding this integration in home assistant it fails straight away (no error message or config options). In the log is the following:

``` python
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 339, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/usr/src/homeassistant/homeassistant/components/slimproto/__init__.py", line 19, in async_setup_entry
    await slimserver.start()
  File "/usr/local/lib/python3.9/site-packages/aioslimproto/server.py", line 60, in start
    self._socket_servers += await self.cli.start()
  File "/usr/local/lib/python3.9/site-packages/aioslimproto/cli.py", line 120, in start
    await asyncio.start_server(
  File "/usr/local/lib/python3.9/asyncio/streams.py", line 94, in start_server
    return await loop.create_server(factory, host, port, **kwds)
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1506, in create_server
    raise OSError(err.errno, 'error while attempting '
OSError: [Errno 98] error while attempting to bind on address ('0.0.0.0', 9091): address in use
```
Note that on an earlier attempt, the port in use was actually 3483. I'm running home assistant core 2022.6.4 in a docker container, network set to host, can't find any other binds for port 3483 so not sure why that one failed and afterwards worked, but 9091 is in use by another process (transmission RPC). Can you make the port editable?

Just checked, and looks like these are the ports used (together with 9090) by the slimserver.

**Comment by sophof** [comment]
I'm not sure what you mean, there doesn't seem to be any option to just install slimproto?
The following is from the d[ocumentation](https://www.home-assistant.io/integrations/slimproto) that I followed:

> The SlimProto integration allows you to control a [Squeezebox](https://en.wikipedia.org/wiki/Squeezebox_%28network_music_player%29) music player directly from Home Assistant, without the need for an external Media Server such as Logitech Media Server. This lets you control Squeezebox hardware like the Classic, Transporter, Duet, Boom, Radio, and Touch and software players like [Squeezelite](https://github.com/ralph-irving/squeezelite), [PiCorePlayer](https://www.picoreplayer.org/) or [Max2Play](https://www.max2play.com/en/). For the real DIY enthusiast, [there even is a player for the ESP32](https://github.com/sle118/squeezelite-esp32).

adding the integration then creates the error above, which indeed seems to stem from the server, but I can't specify I don't need it.

**Comment by marcelveldt** [comment]
1) Uninstall or temporary power off the Logitech Media Server you have already running.
2) Can you move that Transmission RPC to another port ?

I can add a bit of code to auto take another free port if that is needed.

**Comment by sophof** [comment]
I don't have a LMS running, all I have is squeezelite installed on a raspberry pi zero with an amp which is installed inside a small speaker (I used it as a spotify alarm clock for a while). What I understood is that I need this integration next to play music/sounds/tts via music assistant on this remote speaker, is that wrong?

**Comment by marcelveldt** [comment]
Ah yes that is exactly what you need. I made the assumption you already had Logitech Media Server installed and that poer collides with the slimproto integration.

Well, at this time the ports (3484, 9090 and 9091) are hardcoded. 3484 is static for slimproto and can not be changed while the 90xx ports are changeable. I will have to adjust the code to auto pick another free port if the current one(s) is occupied.

Can you test if temporary disabling that other app/service/thing you have running on port 9091 (transmission?) works in getting the slimproto working ?

**Comment by sophof** [comment]
I disabled transmission, restarted hass and then it works as expected 👍 My device was added immediately and I could play spotify via the music assistant.

**Comment by marcelveldt** [comment]
This issue is fixed in the code and will be available in the 2022.7.0 or 2022.7.1 release of HA.
